src/git/git.go

```diff
@@ -104,5 +462,25 @@ func New(log Logger, printElapsedTime ElapsedTime, url, username, password, dest
 
 	gitLabVersioning.repo = repo
 
+	if err := gitLabVersioning.getBranchPointedToHead(); err != nil {
+		return nil, err
+	}
+
+	if err := gitLabVersioning.getCommitHistory(); err != nil {
+		return nil, err
+	}
+
+	if err := gitLabVersioning.getMostRecentCommit(); err != nil {
+		return nil, err
+	}
+
+	if err := gitLabVersioning.getAllTags(); err != nil {
+		return nil, err
```

Comment:
should we avoid return error here when it is the first release? Because getAllTags() seems to return "no tags found." err and the next method getMostRecentTag has a "if not tags use starts as 0.0.0" @esequielvirtuoso

Reply:
I'll review this one. Thanks!